core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -0,0 +1,357 @@
+   */
+  final def mergeAllPar[E, A, B](in: Iterable[IO[E, A]])(zero: B)(f: (B, A) => B): IO[E, B] =
+    ZIO.mergeAllPar(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.raceAll]]
+   */
+  final def raceAll[E, A](zio: IO[E, A], ios: Iterable[IO[E, A]]): IO[E, A] = ZIO.raceAll(zio, ios)
+
+  /** See [[scalaz.zio.ZIO.reduceAll]]
+   */
+  final def reduceAll[E, A](a: IO[E, A], as: Iterable[IO[E, A]])(f: (A, A) => A): IO[E, A] =
+    ZIO.reduceAll(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.reduceAllPar]]
+   */
+  final def reduceAllPar[E, A](a: IO[E, A], as: Iterable[IO[E, A]])(f: (A, A) => A): IO[E, A] =
+    ZIO.reduceAllPar(a, as)(f)
+
+  /** See [[scalaz.zio.ZIO.require]]
+   */
+  final def require[E, A](error: E): IO[E, Option[A]] => IO[E, A] =
+    ZIO.require(error)
+
+  /** See [[scalaz.zio.ZIO.reserve]]
+   */
+  def reserve[E, A, B](reservation: IO[E, Reservation[Any, E, A]])(use: A => IO[E, B]): IO[E, B] =
+    ZIO.reserve(reservation)(use)
+
+  /** See [[scalaz.zio.ZIO.runtime]]
+   */
+  final def runtime: IO[Nothing, Runtime[Any]] = ZIO.runtime
+
+  /** See [[scalaz.zio.ZIO.succeed]]
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** See [[scalaz.zio.ZIO.succeedLazy]]
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** See [[scalaz.zio.ZIO.supervise]]
+   */
+  final def supervise[E, A](zio: IO[E, A]): IO[E, A] =
+    ZIO.supervise(zio)
+
+  /** See [[scalaz.zio.ZIO.supervised]]
+   */
+  def supervised[E, A](zio: IO[E, A]): IO[E, A] =
+    ZIO.supervised(zio)
+
+  /** See [[scalaz.zio.ZIO.supervisedWith]]
+   */
+  final def superviseWith[E, A](zio: IO[E, A])(supervisor: IndexedSeq[Fiber[_, _]] => IO[Nothing, _]): IO[E, A] =
+    ZIO.superviseWith(zio)(supervisor)
+
+  /** See [[scalaz.zio.ZIO.suspend]]
+   */
+  final def suspend[E, A](io: => IO[E, A]): IO[E, A] =
+    ZIO.suspend(io)
+
+  /** See [[scalaz.zio.ZIO.swap]]
+   */
+  final def swap[E, A, B](implicit ev: Any <:< (A, B)): IO[E, (B, A)] =
+    ZIO.swap
+
+  /** See [[scalaz.zio.ZIO.uninterruptible]]
+   */
+  final def uninterruptible[E, A](zio: IO[E, A]): IO[E, A] =
+    ZIO.uninterruptible(zio)
+
+  /** See [[scalaz.zio.ZIO.uninterruptibleMask]]
+   */
+  final def uninterruptibleMask[E, A](k: ZIO.InterruptStatusRestore => IO[E, A]): IO[E, A] =
+    ZIO.uninterruptibleMask(k)
+
+  /** See [[scalaz.zio.ZIO.unsandbox]]
+   */
+  final def unsandbox[E, A](v: IO[Cause[E], A]): IO[E, A] = ZIO.unsandbox(v)
+
+  /** See [[scalaz.zio.ZIO.when]]
+   */
+  final def when[E](b: Boolean)(zio: IO[E, _]): IO[E, Unit] =
+    ZIO.when(b)(zio)
+
+  /** See [[scalaz.zio.ZIO.whenM]]
+   */
+  final def whenM[E](b: IO[E, Boolean])(zio: IO[E, _]): IO[E, Unit] =
+    ZIO.whenM(b)(zio)
+
+  /** See [[scalaz.zio.ZIO._1]]
+   */
+  final def _1[E, A, B](implicit ev: Any <:< (A, B)): IO[E, A] = ZIO._1
+
+  /** See [[scalaz.zio.ZIO._2]]
+   */
+  final def _2[E, A, B](implicit ev: Any <:< (A, B)): IO[E, B] = ZIO._2
```

Comment:
This function only makes sense when `R` != `Any` and should be deleted from `IO`.